File: Swexpertacademy/cpp/3947.cpp
#include <iostream>
#include <fstream>
using namespace std;
#define INF -987654321
class 
int main(){
	ifstream in("input.txt");

	int T;
	in>>T;
	for(int tc=1; tc<=T; tc++){
		int N, M;
		in>>N>>M;
		int **costs = new int*[M];
		for(int i=0; i<M; i++) {
			costs[i] = new int[3];
			in>>costs[i][0]>>costs[i][1]>>costs[i][2];
		}

	}
	return 0;
}

File: Swexpertacademy/python/3947.py
import sys
import heapq
sys.stdin = open('input.txt')

for tc in range(1, int(input())+1):
    N, M = map(int, input().split())
    costs = {}
    for i in range(M):
        s, e, c = map(int, input().split())
        if s in costs:
            costs[s].append((e, c))
        else:
            costs[s] = [(e, c)]
    dist = []
    print(costs)